pallets/registry/src/lib.rs

```diff
@@ -361,16 +353,16 @@ impl<T: Config> VerifierRegistry for Pallet<T> {
 		// Caller is the owner of the registry
 		Owner::<T>::insert(id.clone(), caller);
 
-		Ok(id)
+		id
 	}
 
 	/// Mint of a non-fungible token
 	fn mint(
-		caller: Self::AccountId,
-		owner_account: Self::AccountId,
-		asset_id: AssetId<Self::RegistryId, T::TokenId>,
-		asset_info: Self::AssetInfo,
-		mint_info: Self::MintInfo,
+		caller: T::AccountId,
+		owner_account: T::AccountId,
+		asset_id: AssetId<T::RegistryId, T::TokenId>,
+		asset_info: T::AssetInfo,
+		mint_info: MintInfo<T::Hash, T::Hash>,
```

Comment:
Perhaps use the alias `SystemHashOf<T>` so that to be coherent with line 280

Reply:
I would also sugest we remove SystemHashOf since its mostly just T:Hash. 
I would suggest we use alias for something that is not super straight forward to understand

Reply:
lets keep that for our nft studio refactor 😉